### DISCUSSION TITLE
remult-admin

### DISCUSSION THREAD

**Discussion by jc955** [noam-honig/radweb#374] [General]
[remult-admin/README.md](https://github.com/remult/remult/blob/master/projects/remult-admin/README.md)
Hi, when I was checking the repository, I found remult-admin, is it a pocketbase-like backend admin interface? Is it possible to add, delete, change, view data, manage permissions, data relationships, logs, etc.? If so, that's awesome and has a lot of potential!

However, the live demo link doesn't work and there is very little documentation.

**Comment by noam-honig** [comment]
Hi @kk7771, Yes - remult admin does exactly that - it's now included with remult itself, and can be easily activated using `admin:true` in `remultExpress` or fastify or....

Checkout these release notes:
https://github.com/remult/remult/releases/tag/v0.25.5

You can also check it out live in the CRM demo:
https://crm-demo.up.railway.app/

sign in and select this menu
![image](https://github.com/remult/remult/assets/16635859/b6f1a409-1839-4fa1-b7d1-37aad4be7eee)

**Comment by noam-honig** [comment]
I've also updated the readme now :)

**Reply by jc955** [reply]
I tried it pretty great, but the base functionality is limited. In the future, will it support complex backend management like PocketBase, with fine-grained control via UI.  I think it's a great trick～

[pocketbase demo](https://pocketbase.io/demo/) This is the back-end administration interface of PocketBase

**Reply by noam-honig** [reply]
Yes - we plan to add features to remult-admin based on usage and user requirements.

There are some differences between pocket base and remult-admin in the sense that many operations, like adding fields, rules etc... in remult are done in the code itself.

If you have specific operations that are more of a priority to you, please let me know

Also - it's a relatively simple react project, and you are invited to contribute to it.

**Reply by jc955** [reply]
My starting point was to quickly build an MVP, and pockerbase comes with a lot of integrated functionality

1. easy to manage and view the relationship & permissions of each Collections
2. view development logs for quick troubleshooting
3. Elegant Auth providers access
4. Automatic backup.
5. CRUD API
6. Easy deployment

It can quickly complete an MVP, from 0 to 1, which may be the reason why it is loved by many people. Of course, I know that remult is compatible with many frameworks, but there may be some limitations. I believe that many people use remult, but also want to quickly develop high-quality products.

**Reply by noam-honig** [reply]
I believe that remult is a great choice when you need to get an MVP from 0 to 1 quickly.

With remult, we believe everything should be in the code in an easy-to-read, yet flexible way.

You can easily manage and view relationships, & permissions in the entity itself - in your code.
The CRUD API is automatic and fully flexible, with a fully typed end-to-end query language that keeps you safe as you develop.

With remult you actually share code between the frontend and the backend - so you can get a consistent validation and authorization system between the UI and the backend.

The approach of doing it all from code, vs defining it in a UI, gives you the full power of code (refactoring etc...) and easy version control using standard tools (git).

Another advantage is that with remult you have full control over the backend, and it's code. You use the same language (typescript) for the backend and the same data access logic (using repositories) on the backend and the frontend.

Some of the concerns you've mentioned are external to remult, simply because there are great existing solutions that one can use - we do provide tutorials on how to build an end-to-end application, including authentication, and deployment:

https://remult.dev/docs/#follow-a-step-by-step-tutorial
Or watch them on videos:
https://www.youtube.com/playlist?list=PLlcnBwFkuOn166nXXxxfL9Hee-1GWlDSm

One more advantage of remult is that you can use it with many different databases (sqlite, postgres, MySQL, SQL server, oracle mongo etc...), and many different backend frameworks (express, fastify, nest, holo etc...) and deploy it to any nodejs server or serverless architecture.

**Reply by jycouet** [reply]
I roger everything that @noam-honig wrote 👍

---

And would like to add a few things _(in random order)_: 
- in remult `admin` you can see the diagram of your entities. So if you like visuals _(like me)_, it's kinda nice to check your code on this page. And if you see a missing part: Edit your code & refresh the page and voila 🎉

- Even if I love visual, in "real" projects, it's kinda hard to track things out. 😅
![Capture d’écran 2024-03-20 à 11 43 57](https://github.com/remult/remult/assets/5312607/5b3311be-f200-475f-be66-b16c4a470bc2)

- I believe also that code as source of truth is very powerful as you can spin up different envs, tests, ... very easily.

- We have some ideas to help you generating a few bits of your code with UI _(`remult-studio` or another name 😉)_ let's see how far we can push things out. I personally see this as a helper, but in the end, you have code, and **you own the code**. => This is KEY.

- `remult` is agnostic in a lot of ways and that's great, it's a rough diamond. If you're looking for something more opinionated, I started `remult-kit`, which comes with predefined choices like Lucia for authentication, Postgres for the database, sveltekit, ... Coming with modules out of the box (auth, storage, ...)! _(I'm using this for like 6 months, and I'm in the process of open sourcing it... doc, ... All this takes time 😉)_

---

Exiting to see all possibilities of `remult` 🎉🎉🎉

**Reply by jc955** [reply]
Thank you for your enthusiastic replies, I couldn't agree with you more. This is what makes remult so powerful and flexible. remult is great.
I hope remult gets better and better 👏

**Reply by noam-honig** [reply]
We're working on it :)
